src/index.js

```diff
@@ -141,12 +143,14 @@ async function main() {
         auth: user.jwt,
       });
     }
-    if (purgePictrsOlderThanDays && pictrsServerApiToken && pictrsUrl) {
-      console.log(`Purging images older than ${purgePictrsOlderThanDays} days`);
-      await purgePictrs(pool);
+    if (l < purgePageSize) {
```

Comment:
<= ?

Reply:
<

If l = PURGE_BATCH_SIZE then that means we hit the limit during the last query and there's still more work to do. Once l < PURGE_BATCH_SIZE then we know we've gotten everything and are ready for the sleep.